### DISCUSSION TITLE
Locking view or accurate mark positions

### DISCUSSION THREAD

**Discussion by txtyash** [gdh1995/vimium-c#826] [Ideas]
So when we resize a browser window, let's say the current content in the view
starts with a webpage content topic named "Headlines", then after resizing,
now the content in the view would be starting with some other area in the
webpage say "Our newsletter".

This happens with all the browsers. I thought marks could help me with this 
problem so that marking the current view when the browser is half width of 
my screen then after making the browser fullscreen I could bring back that
view with revisiting that mark but that didn't work.

I'm sorry that I'm trying to solve a problem with marks that marks probably
weren't intended for but not only could this be a refinement for marks but 
maybe I could get to know if it's possible to lock the view in browsers with
any js code so that I could look into making an extension.

**Comment by gdh1995** [comment]
Well it's a very difficult problem, and up to now `Marks` can not solve it - it just remember scrolling positions (pixel offsets).

It's not easy to write code to find a line to scroll back to after resizing, although it's indeed easy for human beings.

I have no idea about how to do so, except "the one to find header titles".

**Reply by txtyash** [reply]
IG we can access the size of browser window & get the scrollbar length and position so we can compute something mathematically that would relatively change the viewport to that position...
WDYT?

**Reply by gdh1995** [reply]
Marks should have tried to remember and then restore scrolling position, but:
1. some web pages don't support programmed scrolling well during their loading process
2. if a window changes its size, then it's totally impossible to calculate a new scrolling position from the only data of "old scrolling position and window size"
